File: docs/syntax-tests/syntax-test-2.go
package main

import "fmt"

type Widget struct {
	Name   string
	Active bool
	Count  int
}

func (w Widget) Label() string {
	return fmt.Sprintf("%s:%d", w.Name, w.Count)
}

func main() {
	widget := Widget{Name: "syntax", Active: true, Count: 4}
	fmt.Println(widget.Label())
}

File: docs/syntax-tests/syntax-test-2.rs
#[derive(Debug, Clone)]
struct Widget<'a> {
    name: &'a str,
    count: usize,
}

impl Widget<'_> {
    fn label(&self) -> String {
        format!("{}:{}", self.name, self.count)
    }
}

fn main() {
    let widget = Widget {
        name: "syntax",
        count: 6,
    };

    println!("{}", widget.label());
}
